From 9759b5fc04cdaec21405569663f6f7e8ed58eaf3 Mon Sep 17 00:00:00 2001
From: Ruslan Bay <ruslanbey@proton.me>
Date: Thu, 26 Feb 2026 17:21:28 +0100
Subject: [PATCH 37/56] media: intel/ipu4: Update to new v4l2_subdev_state pad
 accessors

Replace deprecated v4l2_subdev_get_try_format(), v4l2_subdev_get_try_crop(), and
v4l2_subdev_get_try_compose() with the new v4l2_subdev_state_get_format/crop/compose()
accessor API for accessing pad try state.
---
 drivers/media/pci/intel/ipu-isys-subdev.c | 24 ++++++-----------------
 1 file changed, 6 insertions(+), 18 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-isys-subdev.c b/drivers/media/pci/intel/ipu-isys-subdev.c
index ab7500529abe..f294c0ef951f 100644
--- a/drivers/media/pci/intel/ipu-isys-subdev.c
+++ b/drivers/media/pci/intel/ipu-isys-subdev.c
@@ -158,9 +158,7 @@ struct v4l2_mbus_framefmt *__ipu_isys_get_ffmt(struct v4l2_subdev *sd,
 	if (which == V4L2_SUBDEV_FORMAT_ACTIVE)
 		return &asd->ffmt[pad][stream];
 	else
-		return v4l2_subdev_get_try_format(
-							 sd,
-							 cfg, pad);
+		return v4l2_subdev_state_get_format(cfg, pad);
 }
 
 struct v4l2_rect *__ipu_isys_get_selection(struct v4l2_subdev *sd,
@@ -180,13 +178,9 @@ struct v4l2_rect *__ipu_isys_get_selection(struct v4l2_subdev *sd,
 	} else {
 		switch (target) {
 		case V4L2_SEL_TGT_CROP:
-			return v4l2_subdev_get_try_crop(
-							       sd,
-							       cfg, pad);
+			return v4l2_subdev_state_get_crop(cfg, pad);
 		case V4L2_SEL_TGT_COMPOSE:
-			return v4l2_subdev_get_try_compose(
-								  sd,
-								  cfg, pad);
+			return v4l2_subdev_state_get_compose(cfg, pad);
 		}
 	}
 	WARN_ON(1);
@@ -822,17 +816,11 @@ int ipu_isys_subdev_open(struct v4l2_subdev *sd, struct v4l2_subdev_fh *fh)
 
 	for (i = 0; i < asd->sd.entity.num_pads; i++) {
 		struct v4l2_mbus_framefmt *try_fmt =
-			v4l2_subdev_get_try_format(
-			sd, fh->state,
-			i);
+			v4l2_subdev_state_get_format(fh->state, i);
 		struct v4l2_rect *try_crop =
-			v4l2_subdev_get_try_crop(
-			sd, fh->state,
-			i);
+			v4l2_subdev_state_get_crop(fh->state, i);
 		struct v4l2_rect *try_compose =
-			v4l2_subdev_get_try_compose(
-			sd, fh->state,
-			i);
+			v4l2_subdev_state_get_compose(fh->state, i);
 
 		*try_fmt = asd->ffmt[i][0];
 		*try_crop = asd->crop[i];
-- 
2.51.0

